service/scaladsl/kafka/server/src/main/scala/com/lightbend/lagom/scaladsl/broker/kafka/LagomKafkaComponents.scala

```diff
@@ -16,6 +16,12 @@ trait LagomKafkaComponents extends LagomKafkaClientComponents {
   def lagomServer: LagomServer
   def offsetStore: OffsetStore
 
+  override def topicPublisherName: Option[String] = super.topicPublisherName match {
+    case Some(other) =>
+      sys.error(s"Cannot provide the kafka topic factory as the default topic publisher since a default topic publisher has already been mixed into this cake: $other")
```

Comment:
We introduce the concept of cake here:

http://www.lagomframework.com/documentation/1.3.x/scala/ServiceImplementation.html#wiring-together-a-lagom-application

Though probably not sufficiently well.  The nice thing about using the term cake is that it makes it clear exactly where the problem is in your app and what sort of problem it is - but that does assume that you know what a cake is.